Formula/gnupg.rb

```diff
@@ -13,70 +13,92 @@ class Gnupg < Formula
+  depends_on "homebrew/fuse/encfs" => :optional
+
+  # ssh-import.scm fails during "make check" for sandboxed builds
+  # Reported 1 Mar 2017 https://bugs.gnupg.org/gnupg/issue2980
+  patch :DATA
 
   def install
     args = %W[
       --disable-dependency-tracking
       --disable-silent-rules
       --prefix=#{prefix}
-      --disable-asm
-      --program-suffix=1
+      --sbindir=#{bin}
+      --sysconfdir=#{etc}
+      --enable-symcryptrun
+      --with-pinentry-pgm=#{Formula["pinentry"].opt_bin}/pinentry
     ]
-    args << "--with-libusb=no" if build.without? "libusb-compat"
+
+    args << "--disable-ccid-driver" if build.without? "libusb"
+    args << "--with-readline=#{Formula["readline"].opt_prefix}" if build.with? "readline"
 
     system "./configure", *args
-    system "make"
-    system "make", "check"
 
-    # we need to create these directories because the install target has the
-    # dependency order wrong
-    [bin, libexec/"gnupg"].each(&:mkpath)
     system "make", "install"
 
-    # https://lists.gnupg.org/pipermail/gnupg-devel/2016-August/031533.html
-    inreplace bin/"gpg-zip1", "GPG=gpg", "GPG=gpg1"
+    # "make check" cannot run before "make install"
+    # Reported 1 Mar 2017 https://bugs.gnupg.org/gnupg/issue2979
+    system "make", "check"
 
-    # Although gpg2 support should be pretty universal these days
-    # keep vanilla `gpg` executables available, at least for now.
-    %w[gpg-zip1 gpg1 gpgsplit1 gpgv1].each do |cmd|
-      (libexec/"gpgbin").install_symlink bin/cmd => cmd.to_s.sub(/1/, "")
-    end
+    bin.install "tools/gpgsplit" => "gpgsplit2" if build.with? "gpgsplit"
   end
 
-  def caveats; <<-EOS.undent
-    This formula does not install either `gpg` or `gpgv` executables into
-    the PATH.
+  def post_install
+    (var/"run").mkpath
+  end
 
-    If you simply require `gpg` and `gpgv` executables without explicitly
-    needing GnuPG 1.x we recommend:
-      brew install gnupg2
+  def caveats; <<-EOS.undent
+    Once you run the new gpg2 binary you will find it incredibly
+    difficult to go back to using `gnupg@1.x` or `gnupg@2.0`
+    The new 2.1.x moves to a new keychain format that can't be
+    and won't be understood by the 2.0.x branch or lower.
 
-    If you really need to use these tools without the "1" suffix you can
-    add a "gpgbin" directory to your PATH from your #{shell_profile} like:
+    If you use this formula for a while and decide
+    you don't like it, you will lose the keys you've imported since.
+    For this reason, we strongly advise that you make a backup
+    of your `~/.gnupg` directory.
 
-        PATH="#{opt_libexec}/gpgbin:$PATH"
+    For full details of the changes, please visit:
+      https://www.gnupg.org/faq/whats-new-in-2.1.html
 
-    Note that doing so may interfere with GPG-using formulae installed via
-    Homebrew.
+    If you are upgrading to gnupg 2.1.x from gnupg 2.0.x you should execute:
+      `killall gpg-agent && gpg-agent --daemon`
```

Comment:
Solid chance Mike makes a disapproving face at me here, but if you're going to mass-update people it might be polite to do this automatically in `postinstall` given how much we both know people bother to read caveats.